### PR TITLE
newlib: Split newlib into one generic part and one default syscalls part

### PR DESCRIPTION
**Pull request by gebart**
This PR splits the contents of sys/newlib into one part for the default syscall implementation, and one part for setting up the build environment for building with newlib, regardless of what syscall implementation is used.

This will help some of the MIPS ports which have their own semihosting newlib environment with a syscall implementation.

### CODE REVIEW

File: sys/Makefile.include [hunk 1]
```diff
@@ -65,8 +65,12 @@ ifneq (,$(filter log_%,$(USEMODULE)))
     include $(RIOTBASE)/sys/log/Makefile.include
 endif
 
+ifneq (,$(filter newlib,$(USEMODULE)))
+    include $(RIOTBASE)/sys/Makefile.newlib
+endif
+
 ifneq (,$(filter newlib_syscalls_default,$(USEMODULE)))
-    include $(RIOTBASE)/sys/newlib/Makefile.include
+    include $(RIOTBASE)/sys/newlib_syscalls_default/Makefile.include
```
Comment (kaspar030) [inline sys/Makefile.include]: how about moving this into a subfolder of newlib?

**Comment by gebart** [comment]
- rebased

@kaspar030 do you mind taking a peek at this PR?

File: sys/Makefile.include (hunk 1 above)
Comment (gebart) [inline sys/Makefile.include]: the newlib subfolder is gone. Makefile.newlib is the place for newlib settings.

Comment (gebart) [inline sys/Makefile.include]: which I hope we can move to makefiles/newlib.mk or something in a separate PR.

**Comment by gebart** [comment]
rebased on #4699 as an experiment

**Comment by gebart** [comment]
rebased after #4699 was merged

**Comment by gebart** [comment]
@kaspar030 ping
Murdock is green, are you fine with the changes?

**Comment by neiljay** [comment]
ACK from me

**Comment by miri64** [comment]
@gebart the last Murdock build is pre-release (and Jenkins and Murdock were still activaded). Would you be so kind to rebase.

**Comment by kaspar030** [comment]
Just triggering murdock is enough to build-test a merge into current master.

**Comment by miri64** [comment]
Oh right.

**Pull request merged**